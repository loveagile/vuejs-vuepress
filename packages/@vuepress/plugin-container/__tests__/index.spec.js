const { createApp } = require('@vuepress/core')
const { getFragments } = require('@vuepress/test-utils')
import containerPlugin from '..'

describe('containers', async () => {
  let app

  beforeAll(async () => {
    app = createApp()
    app.options.siteConfig = {
      plugins: [
        [containerPlugin, { type: 'tip' }],
        [containerPlugin, { type: 'warning' }],
        [containerPlugin, { type: 'danger' }],
        [containerPlugin, {
          type: 'slot',
          before: info => `<template slot="${info}">`,
          after: '</template>'
        }],
        [containerPlugin, {
          type: 'v-pre',
          before: '<div v-pre>',
          after: '</div>'
        }]
      ]
    }
    return app.process()
  })

  getFragments(__dirname).forEach(({ name, content: input }) => {
    test(name, () => {
      const { html } = app.markdown.render(input)
      expect(html).toMatchSnapshot()
    })
  })
})
