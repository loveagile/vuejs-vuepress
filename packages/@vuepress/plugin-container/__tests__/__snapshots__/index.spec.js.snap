// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers danger.md 1`] = `
<div class="danger custom-block">
  <p class="custom-block-title">DANGER</p>
  <p>I am a danger</p>
</div>
`;

exports[`containers markdown-slots.md 1`] = `
<template slot="header">
  <h1 id="here-might-be-a-page-title"><a class="header-anchor" href="#here-might-be-a-page-title" aria-hidden="true">#</a> Here might be a page title</h1>
</template>
<ul>
  <li>A Paragraph</li>
  <li>Another Paragraph</li>
</ul>
<template slot="footer">
  <p>Here's some contact info</p>
</template>
`;

exports[`containers tip.md 1`] = `
<div class="tip custom-block">
  <p class="custom-block-title">TIP</p>
  <p>I am a tip</p>
</div>
`;

exports[`containers tip-override.md 1`] = `
<div class="tip custom-block">
  <p class="custom-block-title">提示</p>
  <p>I am a tip</p>
</div>
`;

exports[`containers v-pre.md 1`] = `
<div v-pre>
  <p>I am a v-pre</p>
</div>
`;

exports[`containers warning.md 1`] = `
<div class="warning custom-block">
  <p class="custom-block-title">WARNING</p>
  <p>I am a warning</p>
</div>
`;
